src/heranca_exemplos/marcas_de_veiculos/Base.java

```diff
@@ -12,94 +10,94 @@ public class Base {
 	protected String marca;
 	protected String nome;
 	protected String tipoDeVeiculo;
-	protected int quantidadeDePortas;
-	protected int quantidadeDeRodas;
-	public static int veiculoEmOrdem = 0;
-	protected List<String> veiculosEsquematizados = new ArrayList<String>();
-	protected List<String> veiculosConstruidos = new ArrayList<String>();
-	protected Map<String, String> concessionarias = new HashMap<String, String>();
-	private int quantidadeDeMateriaisDisponiveis = 30, materiais = 8;
-
-	public void esquematizaModelo(String marca, String nome, String tipoDeVeiculo, int quantidadeDePortas,
-			int quantidadeDeRodas) {
-		this.tipoDeVeiculo = tipoDeVeiculo;
-		this.quantidadeDePortas = quantidadeDePortas;
-		this.quantidadeDeRodas = quantidadeDeRodas;
+	protected int kmPorLitro;
+	protected int velocidade;
+	protected int valor;
+	public static int veiculoEmOrdem = 1;
+	protected List<String> veiculos = new ArrayList<String>();
+	protected List<String> concessionarias = new ArrayList<String>();
+	public int quantidadeDeMateriaisDisponiveis = 30, materiais = 8;
+
+	public void criaVeiculo(String marca, String nome, String tipoDeVeiculo, int kmPorLitro, int velocidade,
```

Comment:
Apenas por curiosidade, dependendo do padrão de projeto pode se usar pNomeParametro, e evita se usar o this, então no seu construtor, poderia ser assim pMarca, pNome... Mas as duas maneiras estão corretas e depende do que for acordado no projeto.